src/indicators/flooding/glofas/utils.py

```diff
@@ -303,87 +307,84 @@ def get_groups_above_threshold(
     return [group for group in groups if group[1] - group[0] >= min_duration]
 
 
-def get_glofas_activations(da_glofas, thresh, ndays):
-    vals = da_glofas.values
-    groups = get_groups_above_threshold(vals, thresh, ndays)
-    df_glofas_act = pd.DataFrame(groups, columns=["start_index", "end_index"])
-    df_glofas_act["num_days"] = (
-        df_glofas_act["end_index"] - df_glofas_act["start_index"]
-    )
-    df_glofas_act["start_date"] = df_glofas_act["start_index"].apply(
-        lambda x: da_glofas.time[x].values
-    )
-    df_glofas_act["end_date"] = df_glofas_act["end_index"].apply(
-        lambda x: da_glofas.time[x].values
+def get_dates_list_from_dataset(
+    da: xr.DataArray, threshold: float, min_duration: int = 1
+) -> List[np.datetime64]:
+    groups = get_groups_above_threshold(
+        observations=da.to_masked_array(),
+        threshold=threshold,
+        min_duration=min_duration,
     )
-    return df_glofas_act
+    return [da.time[group[0] + min_duration - 1].data for group in groups]
 
 
-def get_detection_stats(df_glofas, df_impact, buffer_before, buffer_after):
-    TP = 0
+def get_detection_stats(
```

Comment:
Great to have this here, and implemented so much simpler than I had done before!